File: mini-commerce/src/main/java/me/hero/minicommerce/item/service/dto/ModifiedItemDto.java
package me.hero.minicommerce.item.service.dto;

import lombok.Value;
import me.hero.minicommerce.item.domain.Item;

@Value
public class ModifiedItemDto {
  private final Long id;
  private final String name;
  private final Long price;

  public ModifiedItemDto(Item item) {
    this.id = item.getId();
    this.name = item.getName();
    this.price = item.getPrice();
  }
}

File: mini-commerce/src/main/kotlin/me/hero/minicommerce/item/application/port/in/dto/ModifiedItemDto.kt
package me.hero.minicommerce.item.application.port.`in`.dto

import me.hero.minicommerce.item.domain.Item

data class ModifiedItemDto(
    val id: Long,
    val name: String,
    val price: Long,
) {
    constructor(item: Item) : this(
        id = item.id!!,
        name = item.name,
        price = item.price,
    )
}
